ecl/hthor/hthor.cpp

```diff
@@ -10358,6 +10359,597 @@ void CHThorExternalActivity::stop()
 }
 
 
+//=====================================================================================================
+
+CHThorNewDiskReadBaseActivity::CHThorNewDiskReadBaseActivity(IAgentContext &_agent, unsigned _activityId, unsigned _subgraphId, IHThorDiskReadBaseArg &_arg, IHThorCompoundBaseArg & _segHelper, ThorActivityKind _kind, IPropertyTree *_node)
+: CHThorActivityBase(_agent, _activityId, _subgraphId, _arg, _kind), helper(_arg), segHelper(_segHelper)
+{
+    helper.setCallback(this);
+    expectedDiskMeta = helper.queryDiskRecordSize();
+    projectedDiskMeta = helper.queryProjectedDiskRecordSize();
+    readerOptions.setown(createPTree());
+    /*
+    if (_node)
+    {
+        const char *recordTranslationModeHintText = _node->queryProp("hint[@name='layoutTranslation']/@value");
+        if (recordTranslationModeHintText)
+            recordTranslationModeHint = getTranslationMode(recordTranslationModeHintText);
+    }
+    */
+}
+
+CHThorNewDiskReadBaseActivity::~CHThorNewDiskReadBaseActivity()
+{
+    close();
+}
+
+void CHThorNewDiskReadBaseActivity::ready()
+{
+    CHThorActivityBase::ready();
+
+    grouped = false;
+    opened = false;
+    offsetOfPart = 0;
+    partNum = (unsigned)-1;
+
+    resolveFile();
+
+    fieldFilters.kill();
+    segHelper.createSegmentMonitors(this);
+}
+
+void CHThorNewDiskReadBaseActivity::stop()
+{
+    close();
+    CHThorActivityBase::stop();
+}
+
+void CHThorNewDiskReadBaseActivity::resolveFile()
+{
+    OwnedRoxieString fileName(helper.getFileName());
+    mangleHelperFileName(mangledHelperFileName, fileName, agent.queryWuid(), helper.getFlags());
+    if (helper.getFlags() & (TDXtemporary | TDXjobtemp))
+    {
+        StringBuffer mangledFilename;
+        mangleLocalTempFilename(mangledFilename, mangledHelperFileName.str());
+        tempFileName.set(agent.queryTemporaryFile(mangledFilename.str()));
+        logicalFileName = tempFileName.str();
+        gatherInfo(NULL);
+    }
+    else
+    {
+        ldFile.setown(resolveLFNFlat(agent, mangledHelperFileName.str(), "Read", 0 != (helper.getFlags() & TDRoptional)));
```

Comment:
perhaps not new, but wondering what resolveLFNFlat does vs other types of files?